### PR TITLE
Use one `fsevent_watch` process per listener instead of one per dir

### PR DESCRIPTION
**Pull request by tgxworld**
Related: https://github.com/rails/rails/issues/26158, https://github.com/rails/rails/issues/37269

This is an attempt to resolve the problem of too many `fsevent_watch` events being spun up on OSX since the number of processes increases with the number of directories being watched.

Screenshot 2019-12-02 at 2 20 28 PM

### CODE REVIEW

**Comment by tgxworld** [comment]
cc @ioquatix @rymai Can I get your thoughts regarding this before I look into adding the specs? Thank you!

**Comment by ioquatix** [comment]
Yeah, we should definitely fix this.

File: spec/lib/listen/adapter/darwin_spec.rb [hunk 1]
```diff
@@ -69,130 +70,4 @@
       it { should eq 1234 }
     end
   end
-
-  describe 'multiple dirs' do
```
Comment (tgxworld) [inline spec/lib/listen/adapter/darwin_spec.rb]: Specs are no longer relevant and there is already high level acceptance test that runs for OSX on CI.

File: lib/listen/adapter/darwin.rb [hunk 2]
```diff
@@ -35,59 +35,45 @@ def self.usable?
...
-      def _process_event(dir, event)
-        _log :debug, "fsevent: processing event: #{event.inspect}"
-        event.each do |path|
-          new_path = Pathname.new(path.sub(%r{\/$}, ''))
-          _log :debug, "fsevent: #{new_path}"
-          # TODO: does this preserve symlinks?
-          rel_path = new_path.relative_path_from(dir).to_s
-          _queue_change(:dir, dir, rel_path, recursive: true)
+      def _process_changes(dirs)
+        dirs.each do |dir|
```
Comment (tgxworld) [inline lib/listen/adapter/darwin.rb]: I'm aware there are two loops here but for the average use case, this is going to consist of only a couple of directories that change each time.

**Comment by tgxworld** [comment]
@ioquatix Would you mind reviewing this? Thank you so much! 🙇

**Comment by ioquatix** [comment]
I will review it tonight.

**Comment by tgxworld** [comment]
Thank you so much 👍

**Comment by ioquatix** [comment]
Using local branch: https://github.com/guard/listen/pull/471

File: spec/lib/listen/adapter/darwin_spec.rb (hunk 1 above)
Comment (ioquatix) [reply spec/lib/listen/adapter/darwin_spec.rb]: Oh, I see. Thanks.

**Pull request closed without merging**